### PR TITLE
ess fixes and improvements

### PR DESCRIPTION
**Pull request by gdkrmr**
ESS added xref integration with emacs-ess/ESS/pull/475 , this adds a jump handler for spacemacs integration.

### CODE REVIEW

**Comment by gdkrmr** [comment]
should the ESS version get checked?

**Comment by bmag** [comment]
> should the ESS version get checked?

No need, on develop we assume latest packages.

**Comment by gdkrmr** [comment]
I have tested this for a while now and it works like a charm!

**Comment by gdkrmr** [comment]
sorry, I accidentally merged instead of rebasing...

**Comment by gdkrmr** [comment]
I added some extra key bindings for the repl.

**Comment by gdkrmr** [comment]
what is the failing test about?

**Comment by gdkrmr** [comment]
is this going to get merged?

**Comment by sdwolfz** [comment]
@gdkrmr I am not familiar with this layer and it will take a while until I try it out locally (unless another maintainer jumps in). Please have a bit more patience.

**Comment by gdkrmr** [comment]
no problem, I just wanted to make sure that this does not get lost. By now there are some additional changes other than just the x-ref integration.

**Comment by gdkrmr** [comment]
I just pushed a commit that fixes #10964

**Comment by gdkrmr** [comment]
A general suggestion: As there is a dedicated julia layer (#10444) that IMHO is better that the ess julia integration, I would suggest to remove julia support from the ess layer, as in this PR.

**Comment by YiLiu6240** [comment]
I think you could remove [this line](https://github.com/gdkrmr/spacemacs/blob/9f9e57b8e9267173ba22c5e8d50daf3eb48f96ef/layers/%2Blang/ess/packages.el#L60) 
`(add-hook 'ess-mode-hook 'spacemacs/run-prog-mode-hooks)`
as `prog-mode` support [as been added](https://github.com/emacs-ess/ESS/blob/master/lisp/ess-mode.el#L372) to ess.

**Comment by gdkrmr** [comment]
@YiLiu6240 good catch, I tested it and indeed the `prog-mode-hooks` ran two times.

**Comment by gdkrmr** [comment]
@syl20bnr Given than the `ess`-layer broke due to recent changes of ESS I think this should get merged asap.

File: layers/+lang/java/packages.el [hunk 1]
```diff
@@ -347,7 +347,7 @@
 (defun java/post-init-counsel-gtags ()
   (spacemacs/counsel-gtags-define-keys-for-mode 'java-mode))
 
-(defun java/post-init-helm-gtags ()
+(defun ""java/post-init-helm-gtags ()
```
Comment (Isimoro) [inline layers/+lang/java/packages.el]: This change breaks the java layer

Comment (gdkrmr) [reply layers/+lang/java/packages.el]: fixed, no idea how that got there, I did not touch the java layer

**Comment by MilesMcBain** [comment]
@gdkrmr I wonder would it make sense to break out the fix for the `ess-site` deprecation issue into a separate PR so it can be merge more quickly?

ESS in Spacemacs is completely broken without that, while other stuff looks like debatable UI changes.

**Comment by gdkrmr** [comment]
This PR has been around for over 6 months now so I think me splitting up this PR will not accelerate anything. I suggest that people who use the `ess` layer use this branch, I keep it updated with develop and the spacemacs maintainers can merge it whenever they think it is convenient.

**Comment by sdwolfz** [comment]
Thank you :heart:!
Cherry-picked into develop branch, you can safely delete your branch.

In the future please make separate PRs for each fix, it will be easyer to review and merge that way.

**Comment by gdkrmr** [comment]
sorry for the single request with multiple features. My reasoning was that less PRs produce less work for you.

**Pull request merged**